File: apps/myjobhunter/backend/app/schemas/profile/work_history.py
"""WorkHistory schemas — Phase 1 stub."""
import uuid
from datetime import date, datetime

from pydantic import BaseModel


class WorkHistoryRead(BaseModel):
    id: uuid.UUID
    profile_id: uuid.UUID
    company_name: str
    title: str
    start_date: date
    end_date: date | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/myjobhunter/frontend/src/types/work-history/work-history.ts
/**
 * WorkHistory as returned by the backend.
 * Mirrors WorkHistoryResponse in backend/app/schemas/profile/work_history_response.py.
 */
export interface WorkHistory {
  id: string;
  user_id: string;
  profile_id: string;
  company_name: string;
  title: string;
  start_date: string;
  end_date: string | null;
  bullets: string[];
  created_at: string;
  updated_at: string;
}
